src/mame/drivers/hp80.cpp

```diff
@@ -0,0 +1,786 @@
+							pixels = 0xfe;
+						} else {
+							pixels = 0;
+						}
+						for (unsigned sub_x = 0; sub_x < 8; sub_x++) {
+							m_bitmap.pix32(row + sub_row , col + sub_x) = m_palette->pen(BIT(pixels , 7));
+							pixels <<= 1;
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+// Vector table (indexed by bit no. in m_int_serv)
+static const uint8_t vector_table[] = {
+	0x04,	// Keyboard
+	0x08,	// Timer 0
+	0x0a,	// Timer 1
+	0x0c,	// Timer 2
+	0x0e	// Timer 3
+};
+
+IRQ_CALLBACK_MEMBER(hp85_state::irq_callback)
+{
+	BIT_SET(m_int_acked , m_top_pending);
+	update_irl();
+	return vector_table[ m_top_pending ];
+}
+
+WRITE8_MEMBER(hp85_state::ginten_w)
+{
+	m_global_int_en = true;
+	update_irl();
+}
+
+WRITE8_MEMBER(hp85_state::gintdis_w)
+{
+	m_global_int_en = false;
+	update_irl();
+}
+
+READ8_MEMBER(hp85_state::keysts_r)
+{
+	uint8_t res = 0;
+	if (BIT(m_int_en , IRQ_KEYBOARD_BIT)) {
+		BIT_SET(res , 0);
+	}
+	if (m_kb_pressed) {
+		BIT_SET(res , 1);
+	}
+	if (BIT(m_io_modkeys->read() , 0)) {
+		BIT_SET(res , 3);
+	}
+	if (m_global_int_en) {
+		BIT_SET(res , 7);
+	}
+	return res;
+}
+
+WRITE8_MEMBER(hp85_state::keysts_w)
+{
+	logerror("KEYSTS W=%02x\n" , data);
+	if (BIT(data , 0)) {
+		irq_en_w(IRQ_KEYBOARD_BIT , true);
+	} else if (BIT(data , 1)) {
+		irq_en_w(IRQ_KEYBOARD_BIT , false);
+	}
+	// TODO: beeper
+	if (BIT(data , 7)) {
+		m_kb_flipped = !m_kb_flipped;
+	}
+}
+
+READ8_MEMBER(hp85_state::keycod_r)
+{
+	return m_kb_keycode;
+}
+
+WRITE8_MEMBER(hp85_state::keycod_w)
+{
+	logerror("KEYCOD W=%02x\n" , data);
+	if (data == 1) {
+		irq_w(IRQ_KEYBOARD_BIT , false);
+		m_kb_enable = true;
+	}
+}
+
+READ8_MEMBER(hp85_state::crtc_r)
+{
+	uint8_t res = 0xff;
+
+	// Read from CRT controller (1MA5)
+	switch (offset) {
+	case 0:
+		// CRTSAD: write-only
+		break;
+
+	case 1:
+		// CRTBAD: write-only
+		break;
+
+	case 2:
+		// CRTSTS
+		res = m_crt_sts;
+		break;
+
+	case 3:
+		// CRTDAT
+		res = m_crt_read_byte;
+		break;
+	}
+	logerror("RD @%u = %02x\n" , offset , res);
+	return res;
+}
+
+WRITE8_MEMBER(hp85_state::crtc_w)
+{
+	// Write to CRT controller (1MA5)
+	logerror("WR @%u = %02x\n" , offset , data);
+	uint8_t burst_idx = m_cpu->flatten_burst();
+	switch (offset) {
+	case 0:
+		// CRTSAD
+		if (burst_idx == 1) {
+			m_crt_sad = ((uint16_t)data << 8) | (m_crt_sad & 0xff);
+		} else if (burst_idx == 0) {
+			m_crt_sad = (m_crt_sad & 0xff00) | data;
+		}
+		logerror("SAD=%04x m=%u\n" , m_crt_sad , burst_idx);
+		break;
+
+	case 1:
+		// CRTBAD
+		if (burst_idx == 1) {
+			m_crt_bad = ((uint16_t)data << 8) | (m_crt_bad & 0xff);
+		} else if (burst_idx == 0) {
+			m_crt_bad = (m_crt_bad & 0xff00) | data;
+		}
+		logerror("BAD=%04x m=%u\n" , m_crt_bad , burst_idx);
+		break;
+
+	case 2:
+		// CRTCTL
+		m_crt_ctl = data;
+		if (BIT(m_crt_ctl , CRT_CTL_RD_RQ_BIT)) {
+			video_mem_read();
+		}
+		break;
+
+	case 3:
+		// CRTDAT
+		video_mem_write(data);
+		break;
+	}
+}
+
+WRITE8_MEMBER(hp85_state::rselec_w)
+{
+	membank("rombank")->set_entry(data);
```

Comment:
You shouldn't be doing tagmap lookups like this on-the-fly - use a `required_memory_bank` member to find it at start/validation time.